### ISSUE TITLE
TypeError: Cannot read property 'isConnected' of null

### ISSUE THREAD

**Issue by oliviertassinari** [Blizzard/node-rdkafka#344]
I ended up in the following edge case:
1. I close the producer stream
2. I try to write in the producer stream
3. An error is throw: `TypeError: Cannot read property 'isConnected' of null` from this line: https://github.com/Blizzard/node-rdkafka/blob/22c713b2d9dfbb2b0ef9d42bf2b890c4ed9e4133/lib/producer-stream.js#L121

This is the expected behavior given the producer is cleared:
https://github.com/Blizzard/node-rdkafka/blob/22c713b2d9dfbb2b0ef9d42bf2b890c4ed9e4133/lib/producer-stream.js#L268
It's an edge case, let me know if you want to support it. Eventually it could be simply about raising an explicit error. Hopefully this issue description will help people facing it.

**Comment by oliviertassinari** [comment]
Oh, it's an already existing pattern:
https://github.com/Blizzard/node-rdkafka/blob/3a2d5835602806f364108cc2fae338e6615a98fe/lib/producer.js#L166

**Comment by webmakersteve** [comment]
It shouldn't throw a null pointer, so that needs to be adjusted if it is happening. I would prefer it throwing a more specific exception. So it's definitely a bug!

**Comment by webmakersteve** [comment]
Fixed with https://github.com/Blizzard/node-rdkafka/commit/a864340ef345b721414ec9be6245b3064ddae738

No need for the producer to be set to null. I did it previously because the stream has undefined behavior once the producer internally has been closed, but I also want to provide the flexibility of accessing the internal producer after the stream has been closed (e.g. to get metrics, etc..)

So, I took the set to null out.
